tools/image_database_to_video/image_database_to_video.cc

```diff
@@ -0,0 +1,173 @@
+// BoB robotics includes
+#include "common/macros.h"
+#include "common/path.h"
+#include "common/string.h"
+#include "navigation/image_database.h"
+
+// TBB
+#include "tbb/parallel_for_each.h"
+
+// Third-party includes
+#include "third_party/CLI11.hpp"
+#include "third_party/path.h"
+
+// Standard C++ includes
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+using namespace BoBRobotics;
+using namespace Navigation;
+
+// **TODO**: Allow user to choose output video format
+constexpr const char *VideoExtension = ".mp4";
+constexpr const char *VideoFormat = "mp4v";
+
+bool
+readCSVLine(std::ifstream &ifs, std::string &line,
+            std::vector<std::string> &fields)
+{
+    if (!std::getline(ifs, line)) {
+        return false;
+    }
+
+    strSplit(line, ',', fields);
+
+    // Trim whitespace from fields
+    std::for_each(fields.begin(), fields.end(), strTrim);
+
+    return true;
+}
+
+// Strips out the Filename column from the csv file
+void
+convertCSVFile(const ImageDatabase &inDatabase,
+               const filesystem::path &outPath)
+{
+    std::ifstream ifs{ (inDatabase.getPath() / ImageDatabase::EntriesFilename).str() };
+    BOB_ASSERT(ifs.good());
+    ifs.exceptions(std::ios::badbit);
+
+    std::string line;
+    std::vector<std::string> fields;
+
+    BOB_ASSERT(readCSVLine(ifs, line, fields));
+    const auto found = std::find(fields.cbegin(), fields.cend(), "Filename");
+    const size_t filenameIdx = std::distance(fields.cbegin(), found);
+
+    std::vector<size_t> copyIdx;
+    for (size_t i = 0; i < fields.size() - 1; i++) {
```

Comment:
How come you're looping on ``i < fields.size() - 1`` rather than ``i < fields.size()``?

Also, if you want a totally unnecessary optimisation:
``copyIdex.reserve(fields.size() - 1);``

Reply:
Because we're dropping a column from the CSV file. It probably needs a comment saying this though...